plugins/inputs/windows_event_log/wineventlog/utils.go

```diff
@@ -78,10 +79,21 @@ func CreateQuery(path string, levels []string) (*uint16, error) {
 }
 
 func UTF16ToUTF8Bytes(in []byte, length uint32) ([]byte, error) {
-	i := length
-
-	if length%2 != 0 {
-		i = length - 1
+	// Since Windows server 2022, the returned value of used buffer represents for double bytes char count,
+	// which is half of the actual buffer used by byte(what older Windows OS returns), checking if the length
+	//land on the end of used buffer, if no, double it.
+	var i int
+	if IsTheEndOfContent(in, length) {
+		i = int(length)
+		if i%2 != 0 {
+			i--
+		}
+	} else {
+		log.Printf("W! Buffer used: %d is returning as double byte character count, doubling it for decoding", length)
+		i = int(length) * 2
+	}
+	if i > cap(in) {
```

Comment:
I think technically we can merge this with the above else block. Haven't dealt with math functions in Go, but would be something like
```go
else {
  i = math.Min(cap(in), int(length)*2)
}
```

Reply:
Golang min only works with float64, which is annoying to convert back and forth.

Reply:
Okay that's fine by me. No point making it more confusing.